### ISSUE TITLE
Improve the accessibility of SVG files which are used in the spec

### ISSUE THREAD

**Issue by hayatoito** [WICG/webcomponents#341]
See the context:
- https://github.com/w3c/webcomponents/pull/336
- https://github.com/w3c/webcomponents/pull/340

**Comment by chaals** [comment]
I don't see any reason not to accept PR #340 - it gives a big improvement on accessibility of the SVG, and the description _can_ be used until the image changes. I think we should just deal with that problem if it happens.

**Comment by chaals** [comment]
The SVG files used in the spec can probably all be treated in the same way as composed-trees to improve their accessibility. Note that for anything that doesn't have a _pure_ tree structure, e.g. the results of slotting, it might be harder to understand the image. But I am not sure yet.

**Comment by dglazkov** [comment]
:+1:

**Comment by hayatoito** [comment]
I understand that is improvement on accessibility of the SVG. Let's merge this PR. Thank you for your efforts.

I believe that we can do better in the future.

I'm afraid that the description in PR https://github.com/w3c/webcomponents/pull/340 doesn't help readers to understand what the composed tree is. Readers don't have to try to understand the tree structure, every parent-child relationship, in the image here. That's not essential in understanding what the composed tree is. Also it's hard to maintain.

I think the images should be more _non-normative_.

I appreciate your help to make the spec more friendly. Is there any recommend way to give readers more essential valuable information, instead of explaining every non-essential details in the SVG?

**Comment by chaals** [comment]
> Readers don't have to try to understand the tree structure, every parent-child relationship,
> in the image here. That's not essential in understanding what the composed tree is.

True. I agree that a more /descriptive/ description, instead of trying to be precise about the particular details of this image, would be more useful for most uses of the spec.

> Also it's hard to maintain.

Well, not compared to an accessible image. It took me minutes to write the description. It took me days to get the image more or less workable, and if I get the pattern down right I am afraid it will still take hours. And fixing tools like OminGraffle so they don't make it so had to begin with is likely to take a fair bit of work too…

Describing images for use is something of an art, and I'd welcome others doing it with a bit more style than I put into this particular example. It is also important to do for people who are trying to understand the spec - which includes our own colleagues, not just some theoretical "goodness for the universe".

Anyway, when I get to anther image (I'm working on a fair few other things at the same time) I'll have a go at improving how I do the description, and use that to improve the description of this image...

**Comment by annevk** [comment]
Do we still use these graphics somewhere?

**Comment by hayatoito** [comment]
We no longer use these graphics.
